Ninjathulhu/Ninjathulhu.Game/Entity.cs

```diff
@@ -85,4 +84,12 @@ public Entity()
             Properties = new EntityProperties();
         }
     }
+
+    internal static class TypeExtensions
+    {
+        public static Component MaybeComponent(this Type type)
```

Comment:
I've started writing type-conversion functions that might fail and return null called `MaybeType`.